plugins/reporters/cyclonedx/src/main/kotlin/CycloneDxReporter.kt

```diff
@@ -368,37 +370,41 @@ class CycloneDxReporter : Reporter {
         outputFileExtensions.forEach { fileExtension ->
             val outputFile = outputDir.resolve("$outputName.$fileExtension")
 
-            val bomGenerator = when (fileExtension) {
-                "xml" -> BomGeneratorFactory.createXml(schemaVersion, bom).toXmlString()
-                "json" -> {
-                    // JSON output cannot handle extensible types (see [1]), so simply remove them. As JSON output is
-                    // guaranteed to be the last format serialized, it is okay to modify the BOM here without doing a
-                    // deep copy first.
-                    //
-                    // [1] https://github.com/CycloneDX/cyclonedx-core-java/issues/99.
-                    val bomWithoutExtensibleTypes = bom.apply {
-                        components.forEach { component ->
-                            // Clear the "dependencyType".
-                            component.extensibleTypes = null
-
-                            component.licenses.licenses.forEach { license ->
-                                // Clear the "origin".
-                                license.extensibleTypes = null
+            runCatching {
+                val bomGenerator = when (fileExtension) {
+                    "xml" -> BomGeneratorFactory.createXml(schemaVersion, bom).toXmlString()
+                    "json" -> {
```

Comment:
Should the increase of the nesting maybe be prepared by extracting the json serialization into a functon for readability?